### PR TITLE
Added java solution

### PR DESCRIPTION
**Pull request by stevenyuser**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [X] I have tested my code
- [X] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [X] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-617.mdx [hunk 1]
```diff
@@ -77,5 +77,76 @@ int main()
...
+<JavaSection>
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class LoadBalancing {
+    static class Point {
+        public int x;
+        public int y;
```
Comment (nathangong) [inline solutions/usaco-617.mdx:91]: optional: add a constructor

**Review by nathangong — CHANGES_REQUESTED** [review]

**Comment by nathangong** [comment]
^ overall, other than these small things, clean code!

File: solutions/usaco-617.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-617.mdx:91]: agree

File: solutions/usaco-617.mdx [hunk 2]
```diff
@@ -77,5 +77,76 @@ int main()
...
+        for(int x = 0; x < N; x ++) { // index for x divider, each cow
+            for(int y = 0; y < N; y ++) { // index for y divider, each cow
+                int a = cows[x].x + 1; // line for x divider, just the cow location at index + 1
+                int b = cows[y].y + 1; // line for y divider, just the cow location at index + 1
+
+                // stores each quadrant
+                ArrayList<Point> BL = new ArrayList<Point>();
+                ArrayList<Point> BR = new ArrayList<Point>();
+                ArrayList<Point> TL = new ArrayList<Point>();
+                ArrayList<Point> TR = new ArrayList<Point>();
```
[the remark this answers is not in the document]
Comment (stevenyuser) [reply solutions/usaco-617.mdx:117]: Since you already directly committed it, should I still edit this section?

**Comment by stevenyuser** [comment]
> ^ overall, other than these small things, clean code!

Thanks, it's my first time contributing, I'll make sure to keep these suggestions in mind next time!

File: solutions/usaco-617.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/usaco-617.mdx:117]: uh no i only committed some whitespace stuff, this still exists

File: solutions/usaco-617.mdx [hunk 3]
```diff
@@ -77,5 +77,76 @@ int main()
...
+        Point[] cows = new Point[N];
+        for (int i = 0; i < N; i++) {
+            cows[i] = new Point();
+            cows[i].x = io.nextInt();
+            cows[i].y = io.nextInt();
+        }
+
+        int smallestMax = Integer.MAX_VALUE;
+        for (int x = 0; x < N; x ++) {  // index for x divider, each cow
+            for (int y = 0; y < N; y ++) {  // index for y divider, each cow
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-617.mdx:109]:
```suggestion
        for (int x = 0; x < N; x++) {  // index for x divider, each cow
            for (int y = 0; y < N; y++) {  // index for y divider, each cow
```
cursed

File: solutions/usaco-617.mdx [hunk 4]
```diff
@@ -77,5 +77,76 @@ int main()
...
+                        if (cows[i].y < b) { // bottom
+                            BR.add(cows[i]);
+                        } else { // top
+                            TR.add(cows[i]);
+                        }
+                    }
+                }
+
+                // find max and smallest max
+                int maxCows = Math.max(BL.size(), Math.max(BR.size(), Math.max(TL.size(), TR.size())));
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-617.mdx:137]:
```suggestion
                int maxCows = Math.max(Math.max(BL, BR), Math.max(TL, TR));
```

File: solutions/usaco-617.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Aadit-Ambadkar) [reply solutions/usaco-617.mdx:117]:
```suggestion
                int BL = 0;
                int BR = 0;
                int TL = 0;
                int TR = 0;
```

File: solutions/usaco-617.mdx [hunk 5]
```diff
@@ -77,5 +77,76 @@ int main()
...
+                            BL.add(cows[i]);
+                        } else { // top
+                            TL.add(cows[i]);
+                        }
+                    } else { // right
+                        if (cows[i].y < b) { // bottom
+                            BR.add(cows[i]);
+                        } else { // top
+                            TR.add(cows[i]);
+                        }
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-617.mdx:132]:
```suggestion
                    if (cows[i].x < a) { // left
                        if (cows[i].y < b) { // bottom
                            BL++;
                        } else { // top
                            TL++;
                        }
                    } else { // right
                        if (cows[i].y < b) { // bottom
                            BR++;
                        } else { // top
                            TR++;
                        }
```

File: solutions/usaco-617.mdx [hunk 6]
```diff
@@ -77,5 +77,76 @@ int main()
...
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class LoadBalancing {
+    static class Point {
+        public int x;
+        public int y;
+    }
```
Comment (Aadit-Ambadkar) [inline solutions/usaco-617.mdx:92]:
```suggestion
public class LoadBalancing {
```
no longer needed

**Review by Aadit-Ambadkar — CHANGES_REQUESTED** [review]

File: solutions/usaco-617.mdx (hunk 1 above)
Comment (Aadit-Ambadkar) [reply solutions/usaco-617.mdx:91]: just delete it instead.

Comment (nathangong) [reply solutions/usaco-617.mdx:91]: why? it's still used to store the input

Comment (SansPapyrus683) [reply solutions/usaco-617.mdx:91]: aadit found that arraylists weren't necessary so he just... you know

**Pull request closed without merging**